synapse/handlers/identity.py

```diff
@@ -114,10 +114,11 @@ def threepid_from_creds(self, id_server, creds):
 
         query_params = {"sid": session_id, "client_secret": client_secret}
 
-        url = "https://%s%s" % (
-            id_server,
-            "/_matrix/identity/api/v1/3pid/getValidated3pid",
-        )
+        if not id_server.startswith("http"):
+            # Prefix identity server URL with https
+            id_server = "https://" + id_server
```

Comment:
This is due to the unfortunate difference between the config file letting you specify your protocol, and all `id_server` fields in the spec not letting you (and assuming `https://`).

So if you don't provide `http://` or `https://`, we assume you want `https://`.

Reply:
Actually I'm going to change this to a method argument.

Reply:
Actually it turns out this method is only used by account delegates now which have to specify their protocol. Hooray!